### PR TITLE
Allow adding plugins using a config array.

### PR DESCRIPTION
**Pull request by ADmad**
Refs #16537.

### CODE REVIEW

File: src/Core/PluginCollection.php [hunk 1]
```diff
@@ -78,6 +79,41 @@ public function __construct(array $plugins = [])
         $this->loadConfig();
     }
 
+    /**
+     * Add plugins from config array.
+     *
```
Comment (markstory) [inline src/Core/PluginCollection.php:84]: Might be good to include an example of what the expected data should look like.

File: src/Http/BaseApplication.php [hunk 2]
```diff
@@ -174,6 +174,11 @@ public function getPlugins(): PluginCollection
     public function bootstrap(): void
     {
         require_once $this->configDir . 'bootstrap.php';
+
+        $plugins = include $this->configDir . 'plugins.php';
```
Comment (markstory) [inline src/Http/BaseApplication.php]: Can we make this an optional thing? That will make upgrading simpler as applications won't be broken when they upgrade and we can document how to adopt the 'new' thing.

I recognize that we're going into a major release and breaking changes are 'ok' but I feel that we should still try to make upgrading easier at little to no cost to us.

Comment (ADmad) [reply src/Http/BaseApplication.php]: Done

File: src/Core/PluginCollection.php [hunk 3]
```diff
@@ -78,6 +79,49 @@ public function __construct(array $plugins = [])
...
+     * @return void
+     */
+    public function addFromConfig(array $config): void
+    {
+        $debug = Configure::read('debug');
+        $cli = PHP_SAPI === 'cli';
+
+        foreach (Hash::normalize($config) as $name => $options) {
+            $options = (array)$options;
+            $onlyDebug = $options['onlyDebug'] ?? false;
```
Comment (othercorey) [inline src/Core/PluginCollection.php:103]: This could just be `debug` and `cli`.

Comment (ADmad) [reply src/Core/PluginCollection.php:103]: Adding the `only` prefix makes them more readable and intuitive IMO. The plugin load command will correspondingly have `--only-debug` option.

Comment (othercorey) [reply src/Core/PluginCollection.php:103]: Sounds good. At first `onlyDebug` sounded odd compared to `debugOnly`, but it reads correctly in a sentence. Either way.

Comment (ADmad) [reply src/Core/PluginCollection.php:103]: I did flip flop between having "only" as suffix or prefix but it felt better as prefix. Open to changing it to suffix if that's preferred by everyone.

Comment (markstory) [reply src/Core/PluginCollection.php:103]: :+1: for the only prefix. It makes it clearer what the key does and when it is relevant.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Review by othercorey — APPROVED** [review]

**Comment by ADmad** [comment]
I'll make a separate PR for the plugin load/unload commands.

**Pull request merged**